source/mir/deser/package.d

```diff
@@ -876,7 +876,26 @@ template deserializeValue(string[] symbolTable)
             }
 
             alias Types = T.AllowedTypes;
-            alias contains = Contains!Types;
+            alias contains = ContainsProxied!Types;
+
+            pragma(inline, true) void setValue(T)(T new_value)
```

Comment:
That shouldn't be a common logic. What is a use case?

Reply:
It is a too advanced feature for the Dlang's small community. It may make other features like better @nogc support via @serdeScoped harder to implement. Suche is better @nogc support. Please describe what is your use case. I wonder if we can find another way to support it.

Reply:
do you mean the problem is that this is a local function with scope? I can also make it static and use a ref parameter to take in the value to change.

The function body should have no problem with `@nogc` if using `@serdeProxyCast`. The branch that may potentially allocate is also only compiled in if there is a variant that has types which are proxies to primitives / otherwise unsupported value types of the variant.